server/src/main/java/org/apache/druid/guice/DruidProcessingModule.java

```diff
@@ -157,9 +157,16 @@ private void verifyDirectMemory(DruidProcessingConfig config)
       }
     }
     catch (UnsupportedOperationException e) {
+      log.debug("Checking for direct memory size is not support on this platform: %s", e);
       log.info(
-          "Could not verify that you have enough direct memory, so I hope you do! Error message was: %s",
-          e.getMessage()
+          "Unable to determine max direct memory size. If -XX:MaxDirectMemorySize is set, make sure "
+          + "a) -XX:MaxDirectMemorySize is set to at least [%,d] bytes (25%% of maximum heap size), or "
+          + "b) set druid.processing.buffer.sizeBytes, such that "
+          + "druid.processing.buffer.sizeBytes * (druid.processing.numMergeBuffers[%,d] + druid.processing.numThreads[%,d] + 1) "
+          + "does not exceed the value of -XX:MaxDirectMemorySize",
```

Comment:
optional nit: this is still confusing for me, maybe it could be reworded as

```suggestion
          Unable to determine max direct memory size. If druid.processing.buffer.sizeBytes is explicitly set then make sure to set -XX:MaxDirectMemorySize to at least "druid.processing.buffer.sizeBytes * (druid.processing.numMergeBuffers[%,d] + druid.processing.numThreads[%,d] + 1)" or else set it to at least 25% of maximum jvm heap size.
```